kratos/geometries/tetrahedra_3d_10.h

```diff
@@ -1031,6 +1055,27 @@ template<class TPointType> class Tetrahedra3D10 : public Geometry<TPointType>
         return shape_functions_local_gradients;
     }
 
+    /**
+     * Checks if faces are planar. We iterate for all edges and check
+     * that the sum of 0-2 and 2-1 segments is no bigger than 0-1.
+     *
+     * @return bool faces are planar or not
+     *
+     */
+    bool FacesArePlanar() const
+    {
+        constexpr double tol = 1e-6;
+        for (auto edge : this->GenerateEdges()) {
+            const double a = MathUtils<double>::Norm3(edge.GetPoint(0)-edge.GetPoint(1));
+            const double b = MathUtils<double>::Norm3(edge.GetPoint(1)-edge.GetPoint(2));
+            const double c = MathUtils<double>::Norm3(edge.GetPoint(2)-edge.GetPoint(0));
+            if (b + c > a + c*tol) {
```

Comment:
```suggestion
            if (b + c > a*(1.0+tol) ) {
```

Reply:
the tolerance should be based on the full lenght rather than c, which could be close to 0

Reply:
makes sense, done!